tests/karma.js

```diff
@@ -2151,6 +2155,1028 @@ beforeAll(function (done) {
+      var id = [14, 18, 19]
+      var token = ['yrkoLeS33y1aTya', 'eGAWIbKUImVbtu4', 'JbdgMGkt3Cq6B0u']
+      for (var i = 0; i < file.length; i++) {
+        provider.addInteraction({
+          uponReceiving: 'a GET request for a share',
+          withRequest: {
+            method: 'GET',
+            path: Pact.Matchers.term({
+              matcher: '.*\\/ocs\\/v1\\.php\\/apps\\/files_sharing\\/api\\/v1\\/shares$',
+              generate: '/ocs/v1.php/apps/files_sharing/api/v1/shares'
+            }),
+            query: 'path=' + path[i],
+            headers: {
+              authorization: 'Basic YWRtaW46YWRtaW4=',
+              Origin: origin
+            }
+          },
+          willRespondWith: {
+            status: 200,
+            headers: {
+              'Access-Control-Allow-Origin': '*',
+              'Access-Control-Allow-Headers': accessControlAllowHeaders,
+              'Access-Control-Allow-Methods': 'GET,OPTIONS,POST,PUT,DELETE,MKCOL,PROPFIND,PATCH,PROPPATCH,REPORT'
+            },
+            body: '<?xml version="1.0"?>\n' +
+              '<ocs>\n' +
+              ' <meta>\n' +
+              '  <status>ok</status>\n' +
+              '  <statuscode>100</statuscode>\n' +
+              '  <message/>\n' +
+              '  <totalitems></totalitems>\n' +
+              '  <itemsperpage></itemsperpage>\n' +
+              ' </meta>\n' +
+              ' <data>\n' +
+              '  <element>\n' +
+              '   <id>' + id[i] + '</id>\n' +
+              '   <share_type>3</share_type>\n' +
+              '   <uid_owner>admin</uid_owner>\n' +
+              '   <displayname_owner>admin</displayname_owner>\n' +
+              '   <permissions>1</permissions>\n' +
+              '   <parent/>\n' +
+              '   <expiration/>\n' +
+              '   <token>' + token[i] + '</token>\n' +
+              '   <uid_file_owner>admin</uid_file_owner>\n' +
+              '   <displayname_file_owner>admin</displayname_file_owner>\n' +
+              '   <additional_info_owner/>\n' +
+              '   <additional_info_file_owner/>\n' +
+              '   <path>/' + file[i] + '</path>\n' +
+              '   <item_type>file</item_type>\n' +
+              '   <file_target>/' + file[i] + '</file_target>\n' +
+              '   <name/>\n' +
+              '   <url>http://localhost/oc/s/P4sVlWUvQXwro6z</url>\n' +
+              '   <mail_send>0</mail_send>\n' +
+              '   <attributes/>\n' +
+              '  </element>\n' +
+              ' </data>\n' +
+              '</ocs>'
+          }
+        })
+      }
+    }
+    )
+    .then(() => {
+      const shareids = [14, 18, 19]
+      for (const shareid of shareids) {
+        provider.addInteraction({
+          uponReceiving: 'a DELETE request for a share',
+          withRequest: {
+            method: 'DELETE',
+            path: Pact.Matchers.term({
+              matcher: '.*\\/ocs\\/v1\\.php\\/apps\\/files_sharing\\/api\\/v1\\/shares\\/' + shareid + '$',
```

Comment:
does the id really matter? the response is always the same, so maybe we can get away with just a single interaction that matches all ids

Reply:
this gives multiple interaction error, which can be fixed once the refactoring work comes into progress